src/features/auth/public-keys.ts

```diff
@@ -1,43 +1,57 @@
 import type { RequestHandler } from 'express';
-
-import { sbvrUtils } from '@balena/pinejs';
+import { type permissions, sbvrUtils } from '@balena/pinejs';
 
 import {
 	captureException,
 	handleHttpErrors,
 } from '../../infra/error-handling/index.js';
 import { augmentReqApiKeyPermissions } from '../api-keys/lib.js';
 
+export const defaultGetAuthorizedKeysFn = async (
+	req: permissions.PermissionReq,
+	username: string,
+) => {
+	// Augment with the ability to resolve the user's username for this request only, there's no need
+	// for device keys to have the ability by default. Access to the public key will still be restricted
+	// by `user__has__public_key` so this only affects the ability to resolve the username they apply to
+	req = augmentReqApiKeyPermissions(req, ['resin.user.read']);
+	const data = await sbvrUtils.api.resin.get({
+		resource: 'user__has__public_key',
+		options: {
+			$select: 'public_key',
+			$filter: {
+				user: {
+					$any: {
+						$alias: 'u',
+						$expr: {
+							u: { username },
+						},
+					},
+				},
+			},
+		},
+		passthrough: { req },
+	});
+
+	return data.map((e) => e.public_key).join('\n');
+};
+
+let getAuthorizedKeys = defaultGetAuthorizedKeysFn;
+/**
+ * Set the function used to get the authorized keys for a user.
+ */
+export const setGetAuthorizedKeysFn = (
+	$getAuthorizedKeys: typeof getAuthorizedKeys,
+) => (getAuthorizedKeys = $getAuthorizedKeys);
```

Comment:
Is there any reason to return the operation result here rather than just returning `Promise<void>` ?
```suggestion
) => {getAuthorizedKeys = $getAuthorizedKeys};
```